compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -727,23 +691,20 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     /// type parameter encountered while evaluating `f`. Definitions
     /// are created with the parent provided. If no `parent_id` is
     /// provided, no definitions will be returned.
-    ///
-    /// Presuming that in-band lifetimes are enabled, then
-    /// `self.anonymous_lifetime_mode` will be updated to match the
-    /// parameter while `f` is running (and restored afterwards).
     fn collect_in_band_defs<T>(
```

Comment:
In-band lifetimes were removed, what is this logic for?
Maybe it should be renamed if it cannot be removed.